src/core/probe/common.rs

```diff
@@ -30,6 +30,10 @@ pub(crate) fn get_ebpf_debug() -> bool {
     }
 }
 
+/// Options to act on per-probe features.
 pub(crate) enum ProbeOption {
+    /// Control stack trace retrieval.
     StackTrace(bool),
+    /// Control if an event is being sent.
+    SendEvent(bool),
```

Comment:
I haven't looked deeply at the new version but I'm wondering why you chose to do it this way rather than splitting the hook in two.